test/core/rules/test_cases/L035.yml

```
@@ -0,0 +1,58 @@
+rule: L035
+
+no_redundant_else_null:
+  pass_str: |
+    select
+        case
+            when name like '%cat%' then 'meow'
+            when name like '%dog%' then 'woof'
+        end
+    from x
+
+redundant_else_null:
+  fail_str: |
+    select
+        case
+            when name like '%cat%' then 'meow'
+            when name like '%dog%' then 'woof'
+            else null
+        end
+    from x
+
+  fix_str: |
+    select
+        case
+            when name like '%cat%' then 'meow'
+            when name like '%dog%' then 'woof'
+        end
+    from x
+
+alternate_case_when_syntax:
+  skip: Alternate case when syntax is not parse-able right now
+
+  fail_str: |
+    select
+        case name
+            when like '%cat%' then 'meow'
+            when like '%dog%' then 'woof'
+            else null
+        end
+    from x
```

```suggestion
    select
        case name
            when 'cat' then 'meow'
            when 'dog' then 'woof'
            else null
        end
from x
```

I think using `like` there is actually not valid syntax hence the parse error. I tested it in snowflake and it didn't like it!

---

Thanks :)